### ISSUE TITLE
Fix github actions issues

### ISSUE THREAD

**Issue by billimek** [billimek/billimek-charts#20]
The github actions results in two 'push' events when a pull request is merged which results in one of the two duplicate events failing.  There shouldn't be two push events at the same time when a pull request is merged.

**Comment by billimek** [comment]
As of [this commit](https://github.com/billimek/billimek-charts/blob/b0e4a8fe85a39cc90eb61037d35192edee73dd42/.github/main.workflow), `main.workflow` and as of [this commit](https://github.com/billimek/gh-actions/tree/b24e95cb1cab91941ad8d6ef1d40614dbba49ad6/helm-gh-pages) for `entrypoint.sh`, the solution is working as expected:

* Pull requests are handled in a dedicated pull-request workflow and only operate when a pull request is opened or synchronized
* Push events are filtered to only commit operations on master (not branch deletes, etc)
* Merged PRs have the associated branch automatically deleted

**Comment by billimek** [comment]
There is an outstanding bug with the interaction between github actions, their use of the `GITHUB_TOKEN` special automatic token and inability to automatically publish to a github pages site.  See [this issue](https://github.com/maxheld83/ghpages/issues/1) and [this PR](https://github.com/maxheld83/ghpages/pull/18) for more context & background.  It stands to reason that a fix will make it's way into Github actions and make the requirement to use a dedicated personal access token no longer needed.

**Comment by billimek** [comment]
Additional observations:

* It seems to be necessary to make judicious use of 'filters' for push events and pull-reqeust events so that they are not running when they are not intended to be run.  For example, without proper filters, the push event actions will run for new branch creations, updates to existing branches, branch deletions, etc.  Similar situation for pull requests
* It is not a good idea to run filter checks in parallel - at least for pull requests.  What happens is that when one of the filters fail gracefully, the other parallel filter(s) automatically get canceled.  This registers as a check failure in the pull request.

**Comment by billimek** [comment]
This all works now, so closing.
